src/components/FilterForm/FilterForm.tsx

```diff
@@ -0,0 +1,192 @@
+  trialPhase: Object.keys(trialPhase).filter(option => trialPhase[option]),
+  studyType: Object.keys(studyType).filter(option => studyType[option]),
+});
+
+const SORTING_OPTIONS = [
+  { name: 'matchLikelihood', label: 'Match Likelihood', defaultValue: true },
+  { name: 'distance', label: 'Distance', defaultValue: false },
+  { name: 'savedStatus', label: 'Saved Status', defaultValue: false },
+] as const;
+
+const FilterForm = ({
+  defaultValues,
+  blankValues,
+  fullWidth,
+  fullSearchParams,
+  filterOptions,
+  disabled,
+}: FilterFormProps): ReactElement => {
+  const router = useRouter();
+  const theme = useTheme();
+  const isSmallScreen = useMediaQuery(theme.breakpoints.down('md'));
+  const { handleSubmit, control, reset } = useForm<FilterFormValuesType>({ defaultValues });
+
+  const onSubmit = (data: FilterFormValuesType) =>
+    router.push({
+      pathname: '/results',
+      query: { ...fullSearchParams, ...formDataToFilterQuery(data as FilterFormValuesType) },
+    });
+
+  return (
+    <form onSubmit={handleSubmit(onSubmit)}>
+      <Box bgcolor="grey.200">
+        <Grid columns={8} container spacing={2} px={2} py={fullWidth ? 0 : { md: 2 }} pb={{ xs: 2 }} mt={0}>
+          <Grid
+            item
+            xs={8}
+            sx={{ flexDirection: 'row-reverse', justifyContent: 'space-between', alignItems: 'flex-start' }}
+            display="flex"
+          >
+            <Button onClick={() => reset(blankValues)} disabled={disabled} variant="text">
+              Clear all
+            </Button>
+
+            <FilterAccordion title="Sort By" defaultExpanded disabled={disabled}>
+              <FormControl component="fieldset" disabled={disabled}>
+                {SORTING_OPTIONS.map(({ name, label, defaultValue }) => (
+                  <FormControlLabel
+                    key={name}
+                    control={
+                      <Controller
+                        name={`sortingOptions.${name}`}
+                        defaultValue={defaultValue}
+                        control={control}
+                        render={SortingOptionCheckbox}
+                      />
+                    }
+                    label={label}
+                  />
+                ))}
+              </FormControl>
+            </FilterAccordion>
+          </Grid>
+
+          {filterOptions && (
+            <Grid item xs={8}>
+              <FilterAccordion title="Filter By" defaultExpanded disabled={disabled}>
+                <FormControl component="fieldset" disabled={disabled}>
+                  {(filterOptions?.recruitmentStatus || []).length !== 0 && (
+                    <FilterAccordion title="Recruitment Status" defaultExpanded disabled={disabled}>
+                      {filterOptions.recruitmentStatus.map(({ name, label, count }) => (
+                        <Stack direction="row" justifyContent="space-between" alignItems="center" key={`${name}`}>
```

Comment:
Not anymore since there's no way for `name` to be passed in as undefined (escaping it turns it into the string `"undefined"`), so I'm going to revert the change.